runtime/outer-PEC.js

```diff
@@ -102,10 +105,18 @@ class OuterPEC extends PEC {
           handle.mapToStorage(fromHandle);
         }
         if (missingHandles.length > 0) {
-          error = `Recipe couldn't load due to missing handles [recipe=${recipe0}, missingHandles=${missingHandles.join('\n')}].`;
-        } else {
+          const resolvedRecipe = await new RecipeResolver(this._arc).resolve(recipe0);
```

Comment:
It'd be worth ensuring that RecipeResolver returns an already resolved recipe untouched, and just always running through RecipeResolver, as that will also make obvious choices wrt slots.

Reply:
I looked at doing this before merge and hit some issues that require further investigation, I will file an issue to track as follow-up.